pkg/nrtupdater/node_getter.go

```diff
@@ -34,27 +42,34 @@ func (ng *DisabledNodeGetter) Get(ctx context.Context, nodeName string, opts met
 	return nil, fmt.Errorf("%w", NotConfigured)
 }
 
-type OneshotNodeGetter struct {
-	K8SCli kubernetes.Interface
-	node   *corev1.Node
+type CachedNodeGetter struct {
+	nodes map[string]*corev1.Node
 }
 
-func (ng *OneshotNodeGetter) Get(ctx context.Context, nodeName string, opts metav1.GetOptions) (*corev1.Node, error) {
-	if ng.node != nil {
-		return ng.node, nil
+func NewCachedNodeGetter(k8sInterface kubernetes.Interface, ctx context.Context) *CachedNodeGetter {
+	nodelist, err := k8sInterface.CoreV1().Nodes().List(ctx, metav1.ListOptions{})
+	if err != nil {
+		return nil
+	}
+
+	retVal := &CachedNodeGetter{nodes: make(map[string]*corev1.Node, len(nodelist.Items))}
+	for _, node := range nodelist.Items {
+		retVal.nodes[node.Name] = &node
 	}
 
-	node, err := ng.K8SCli.CoreV1().Nodes().Get(ctx, nodeName, opts)
-	if err != nil {
-		return nil, ConnectionError{Err: err}
+	return retVal
+}
+
+func (ng *CachedNodeGetter) Get(ctx context.Context, nodeName string, _ metav1.GetOptions) (*corev1.Node, error) {
+	if node, found := ng.nodes[nodeName]; found {
+		return node, nil
 	}
-	ng.node = node.DeepCopy()
-	return ng.node, nil
+	return nil, fmt.Errorf("%w", NotFound{NodeName: nodeName})
```

Comment:
I don't think there's need to wrap, I think you can just return `NotFound{NodeName: nodeName}`

Reply:
explained in https://github.com/k8stopologyawareschedwg/resource-topology-exporter/pull/212#discussion_r1319842202 . It's more correct this way.